### PR TITLE
feat: interface v2 protocol

### PR DESCRIPTION
**Pull request by doudou**
Depends on:
- [ ] https://github.com/rock-core/tools-roby/pull/275

The integration is done as to not break the v1 protocol. The v2 protocol marshallers are registered only if using the v2 interface, using the app's setup_interface_v2_protocol hook

The biggest change is a plain copy (without changes) of syskit/shell_commands to syskit/interface/commands. The rest is minimal.

### CODE REVIEW

**Comment by doudou** [comment]
Ping

File: lib/syskit/interface/commands.rb
Comment (wvmcastro) [inline lib/syskit/interface/commands.rb:1]: Is it really without changes? _roby_app/shell_inerface_ and _interface/commands_ have different number of lines. Also, I think `git mv` would really do the job here decreasing the diff size

**Review by wvmcastro — APPROVED** [review]

File: lib/syskit/interface/commands.rb
Comment (jhonasiv) [reply lib/syskit/interface/commands.rb:1]: It seems he added some deployment-related functionalities, like the deployment command and moved some code to functions for restart_deployments

**Review by jhonasiv — APPROVED** [review]

File: lib/syskit/interface/commands.rb
Comment (doudou) [reply lib/syskit/interface/commands.rb:1]: `git mv` is strictly equal to `git rm` and `git add`. Git does not have the concept of "moves" per se. It detects copies and moves by textual comparison.

In this case, I believe the issue is with Github, which does not represent copies and moves.

**Pull request merged**